yt_dlp/extractor/wasdtv.py

```diff
@@ -0,0 +1,165 @@
+        return [{
+            'url': url,
+            'preference': index,
+        } for index, url in enumerate(
+            traverse_obj(thumbnails_dict, (('small', 'medium', 'large'),))) if url]
+
+    def _real_extract(self, url):
+        container = self._get_container(url)
+        stream = traverse_obj(container, ('media_container_streams', 0))
+        media = try_get(stream, lambda x: x['stream_media'][0])
+        if not media:
+            raise ExtractorError('Can not extract media data.', expected=True)
+        media_meta = media.get('media_meta')
+        media_url, is_live = self._get_media_url(media_meta)
+        video_id = media.get('media_id') or container.get('media_container_id')
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(media_url, video_id, 'mp4')
+        self._sort_formats(formats)
+        return {
+            'id': str(video_id),
+            'title': container.get('media_container_name') or self._og_search_title(self._download_webpage(url, video_id)),
+            'description': container.get('media_container_description'),
+            'thumbnails': self._extract_thumbnails(media_meta.get('media_preview_images')),
+            'timestamp': parse_iso8601(container.get('created_at')),
+            'view_count': int_or_none(stream.get('stream_current_viewers' if is_live else 'stream_total_viewers')),
+            'is_live': is_live,
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+    def _get_container(self, url):
+        raise NotImplementedError('Subclass for get media container')
+
+    def _get_media_url(self, media_meta):
+        raise NotImplementedError('Subclass for get media url')
+
+
+class WASDTVStreamIE(WASDTVBaseIE):
+    IE_NAME = 'wasdtv:stream'
+    _VALID_URL = r'https?://wasd\.tv/(?P<id>[^/#?]+)$'
+    _TESTS = [{
+        'url': 'https://wasd.tv/24_7',
+        'info_dict': {
+            'id': '559738',
+            'ext': 'mp4',
+            'title': 'Live 24/7 Music',
+            'description': '24&#x2F;7 Music',
+            'timestamp': int,
+            'upload_date': r're:^\d{8}$',
+            'is_live': True,
+            'view_count': int,
+        },
+    }]
+
+    def _get_container(self, url):
+        nickname = self._match_id(url)
+        channel = self._fetch(f'channels/nicknames/{nickname}', video_id=nickname, description='channel')
+        channel_id = channel.get('channel_id')
+        containers = self._fetch(
+            'v2/media-containers',
+            query={
+                'channel_id': channel_id,
+                'media_container_type': 'SINGLE',
+                'media_container_status': 'RUNNING',
+            },
+            video_id=channel_id,
+            description='running media containers')
+        if not containers:
+            raise ExtractorError(f'{nickname} is offline', expected=True)
+        return containers[0]
+
+    def _get_media_url(self, media_meta):
```

Comment:
Does this have `media_archive_url`? If not, `_get_media_url` is actually same for both the classes

Reply:
Yes, sure, It has media_archive_url